packages/wasm-dpp/src/identity/state_transition/asset_lock_proof/mod.rs

```diff
@@ -0,0 +1,269 @@
+use dpp::dashcore::consensus;
+use dpp::dashcore::hashes::hex::FromHex;
+use dpp::identity::errors::UnknownAssetLockProofTypeError;
+use wasm_bindgen::prelude::*;
+
+use crate::buffer::Buffer;
+use crate::errors::dpp_error::from_dpp_error_ref;
+use crate::identifier::IdentifierWrapper;
+use crate::identity::errors::UnknownAssetLockProofTypeErrorWasm;
+use crate::state_repository::{ExternalStateRepositoryLike, ExternalStateRepositoryLikeWrapper};
+use crate::utils::generic_of_js_val;
+use crate::validation::ValidationResultWasm;
+use crate::{Deserialize, StateTransitionExecutionContextWasm};
+use dpp::identity::state_transition::asset_lock_proof::{
+    AssetLockProof, AssetLockProofType, AssetLockPublicKeyHashFetcher,
+    AssetLockTransactionOutputFetcher, AssetLockTransactionValidator,
+};
+
+#[derive(Deserialize, Clone)]
+#[wasm_bindgen(js_name=AssetLockProof)]
+pub struct AssetLockProofWasm(AssetLockProof);
+
+impl From<AssetLockProof> for AssetLockProofWasm {
+    fn from(v: AssetLockProof) -> Self {
+        AssetLockProofWasm(v)
+    }
+}
+
+impl From<AssetLockProofWasm> for AssetLockProof {
+    fn from(v: AssetLockProofWasm) -> Self {
+        v.0
+    }
+}
+
+impl From<InstantAssetLockProofWasm> for AssetLockProofWasm {
+    fn from(f: InstantAssetLockProofWasm) -> Self {
+        AssetLockProof::Instant(f.into()).into()
+    }
+}
+
+impl From<ChainAssetLockProofWasm> for AssetLockProofWasm {
+    fn from(f: ChainAssetLockProofWasm) -> Self {
+        AssetLockProof::Chain(f.into()).into()
+    }
+}
+
+pub trait AssetLockProofLike {
+    fn to_object(&self) -> Result<JsValue, JsValue>;
+}
+
+#[wasm_bindgen(js_class = AssetLockProof)]
+impl AssetLockProofWasm {
+    #[wasm_bindgen(constructor)]
+    pub fn new(raw_asset_lock_proof: JsValue) -> Result<AssetLockProofWasm, JsValue> {
+        let lock_type = get_lock_type(&raw_asset_lock_proof)?;
+
+        match lock_type {
+            AssetLockProofType::Instant => {
+                Ok(InstantAssetLockProofWasm::new(raw_asset_lock_proof)?.into())
+            }
+            AssetLockProofType::Chain => {
+                Ok(ChainAssetLockProofWasm::new(raw_asset_lock_proof)?.into())
+            }
+        }
+    }
+
+    #[wasm_bindgen(js_name=createIdentifier)]
+    pub fn create_identifier(&self) -> Result<IdentifierWrapper, JsValue> {
+        match &self.0 {
+            AssetLockProof::Instant(instant) => {
+                InstantAssetLockProofWasm::from(instant.to_owned().clone()).create_identifier()
+            }
+            AssetLockProof::Chain(chain) => {
+                ChainAssetLockProofWasm::from(chain.to_owned().clone()).create_identifier()
+            }
+        }
+    }
+
+    #[wasm_bindgen(js_name=toObject)]
+    pub fn to_object(&self) -> Result<JsValue, JsValue> {
+        match &self.0 {
+            AssetLockProof::Instant(instant) => {
+                InstantAssetLockProofWasm::from(instant.to_owned().clone()).to_object()
```

Comment:
There's a bunch of stuff Clippy isn't happy about. Although it's not enabled in the CI yet, I still would fix all the warnings produced by it

Reply:
Fixed